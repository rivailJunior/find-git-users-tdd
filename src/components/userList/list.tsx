import { iUser } from "../../interfaceModel/user";

type iList = {
    loading: boolean;
    users: iUser[]
}
export default function List({ loading, users }: iList) {
    return (
        <div>
            <h3>Lista de Usu√°rios</h3>
            {loading ? (
                <div>Loading...</div>
            ) : (
                <ul data-testid="userList">
                    {users?.map((user, index) => {
                        return <li key={index}> {user.login} </li>;
                    })}
                </ul>
            )}
        </div>
    );
}
