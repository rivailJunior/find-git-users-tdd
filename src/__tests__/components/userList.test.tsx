import { render, screen } from '@testing-library/react';
import List from '../../components/userList/list';
import { userList } from '../../mock/userList';

describe('List component', () => {
    describe('Create Snapshot', () => {
        test('Snapshot with loading active and no user on list', () => {
            const { container } = render(<List loading={true} users={[]} />);
            expect(container.firstChild).toMatchSnapshot();
        });
    })

    describe('When user list length is 0', () => {
        test('Show Component Title', () => {
            render(<List loading={true} users={[]} />);
            expect(screen.getByText(/Lista de usuÃ¡rios/i)).toBeInTheDocument()
        });

        test('Show loading feedback to user', () => {
            render(<List loading={true} users={[]} />);
            expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
        });
    });

    describe('When there are users on list', () => {
        test('Render list with 3 users found', () => {
            render(<List loading={false} users={userList} />);
            expect(screen.getByTestId('userList').children.length).toBe(3);
        });

        test('Remove loading information', () => {
            render(<List loading={false} users={userList} />);
            expect(screen.queryByText(/Loading.../i)).not.toBeInTheDocument();
        });
    })

})