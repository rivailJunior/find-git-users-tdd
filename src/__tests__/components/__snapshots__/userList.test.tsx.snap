// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component Create Snapshot Snapshot with loading active and no user on list 1`] = `
<div>
  <h3>
    Lista de Usu√°rios
  </h3>
  <div>
    Loading...
  </div>
</div>
`;
